tests/test_morphing.py

```diff
@@ -105,3 +105,274 @@ def test_original_weight_gradients(
+    ), " The output differs from expected components"
+
+    Nd = 1
+    Np = 1
+    Ns = 0
+    assert np.allclose(
+        new_morpher.find_components(Ns=Ns, Nd=Nd, Np=Np), np.array([2, 2])
+    ), " The output differs from expected components"
+
+    Nd = 0
+    Np = 0
+    Ns = 1
+    assert np.allclose(
+        new_morpher.find_components(Ns=Ns, Nd=Nd, Np=Np), np.array([4])
+    ), " The output differs from expected components"
+
+    Nd = 2
+    Np = 1
+    Ns = 0
+    assert np.allclose(
+        new_morpher.find_components(Ns=Ns, Nd=Nd, Np=Np),
+        np.array([[2, 0, 2], [1, 1, 2], [0, 2, 2]]),
+    ), " The output differs from expected components"
+
+
+def test_compare_weight_gradients(
+    gs=np.array([[1, 1, 1, 1, 1], [-5, -4, -3, -2, -1]]),
+    gp=None,
+    gd=None,
+    parameter_max_power=[2, 2],
+    this_components=np.array([[4, 0], [3, 1], [2, 2], [1, 3], [0, 4]]),
+    this_basis=np.array([[1, -5], [1, -4], [1, -3], [1, -2], [1, -1]]),
+):
+
+    # Test if the weight gradients output the same with the new input format
+    # Morpher object with the new format of inputs
+    new_morpher = m.PhysicsMorpher(
+        parameter_max_power=parameter_max_power,
+    )
+
+    _ = new_morpher.find_components(Ns=2, Nd=0, Np=0)
+    new_morpher.set_basis(basis_d=gd, basis_p=gp, basis_s=gs)
+
+    # Morpher object with the initial format of inputs
+    old_morpher = m.PhysicsMorpher(parameter_max_power=[2, 2])
+    old_morpher.set_components(this_components)
+    old_morpher.set_basis(basis_numpy=this_basis)
+
+    # calculate and check the output values of the old and new mophing weights.
+    old_weight_gradient = old_morpher.calculate_morphing_weight_gradient(theta=[1, 1])
+    new_weight_gradient = new_morpher.calculate_morphing_weight_gradient(theta=[1, 1])
+
+    assert np.allclose(
+        new_weight_gradient, old_weight_gradient
+    ), "The weight gradient of old inputs and the new inputs are different."
+
+
+def test_evaluate_morphing():
+    # Check if the morpher object created with old and new inputs can execute evaluate_morphing() function
+    old_morpher = m.PhysicsMorpher(
+        parameter_max_power=[4, 4],
+        parameter_range=[
+            (
+                -1.0,
+                1.0,
+            ),
+            (-1.0, 1.0),
+        ],
+    )
+    this_components = np.array([[4, 0], [3, 1], [2, 2], [1, 3], [0, 4]])
+    this_basis = np.array([[1, -5], [1, -4], [1, -3], [1, -2], [1, -1]])
+
+    # Set the powers of components and basis
+    old_morpher.set_components(this_components)
+    old_morpher.set_basis(basis_numpy=this_basis)
+
+    old_value = old_morpher.evaluate_morphing(basis=this_basis)
+
+    # Morpher object with the new format of inputs
+    new_morpher = m.PhysicsMorpher(
+        parameter_max_power=[4, 4],
+        parameter_range=[
+            (
+                -1.0,
+                1.0,
+            ),
+            (-1.0, 1.0),
```

Comment:
Keep tuple style consistent:

```python
parameter_range=[
     (-1.0, 1.0),
     (-1.0, 1.0),
]
```

Reply:
Thank you for your feedback, I changed these two input format in the new commit.